### ISSUE TITLE
[pickers] The field does not work when the format is only composed of escaped characters

### ISSUE THREAD

**Issue by noelleabe** [mui/mui-x#10279]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:
https://codesandbox.io/embed/mui-x-issue-template-forked-gs9ck7

Steps:
1. Create a `DatePicker` or `MobileDatePicker` inside a `LocationProvider` and provide a `dateAdapter`.
2. Add a `format` prop that's completely composed of escaped characters. For example, if using adapterMoment, set `format="[test]"`
3. Observe that no text is displayed in the `DatePicker`.
4. In the `format` prop, add an acceptable token.  For example, if using adapterMoment, set `format="[test] Y"`.
5. Observe that all text is displayed correctly in the `DatePicker`.

### Current behavior 😯

When formatted text in a `DatePicker` or `MobileDatePicker` is all escaped characters, nothing is displayed. When the formatted text uses a token in addition to escaped characters, the text is correctly displayed.

### Expected behavior 🤔

Formatted text that is all escaped characters should be displayed regardless of whether a token exists.

### Context 🔦

I want to display the first and last day of the week in a `DatePicker`. Since a week can start and end in different months or years, no tokens from the value are used. 

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
System:
    OS: macOS 12.3
  Binaries:
    Node: 16.18.1 - ~/.nvm/versions/node/v16.18.1/bin/node
    Yarn: Not Found
    npm: 8.19.2 - ~/.nvm/versions/node/v16.18.1/bin/npm
  Browsers:
    Chrome: 116.0.5845.179
    Edge: 116.0.1938.76
    Safari: 15.4
  npmPackages:
    typescript: ^5.3.0-dev.20230817 => 5.3.0-dev.20230830 
</details>

**Comment by flaviendelangle** [comment]
Hi,

Thanks for your feedback.
What is your use-case for rendering a date fully escaped?
Do we agree that it won't be editable and that the UX might be counter-intuitive? People will have a date rendered just like any editable date, but this one will not be selectable / editable in any way.

**Comment by noelleabe** [comment]
My use case is that when users select a date, I want to highlight the entire week and display the first and last day of the week, not the selected day itself. For example, if a user selects September 19, the DatePicker would display September 17 - September 23. Currently, I'm calculating those dates separately and passing them into the format prop.

**Comment by flaviendelangle** [comment]
Make sense
I'll see how hard it is to fix this problem :+1:

**Comment by noelleabe** [comment]
Thank you so much!

**Comment by flaviendelangle** [comment]
I think that if your input does not allow editing, then it should not be an input.
We have an example [on the doc](https://mui.com/x/react-date-pickers/custom-field/#using-a-button) on how to replace it with a button for example.

But if you do want to keep an input, then the solution proposed here would work fine.
It should be available in the next alpha release.

As for correct editing of the weeks though the fields and the views, maybe one day we will provide a clean week picker :+1:

**Comment by flaviendelangle** [comment]
Closes by #10307
